xapian-letor/ranker/listmle_ranker.cc

```diff
@@ -97,15 +100,16 @@ calculateGradient(vector<FeatureVector> & sorted_feature_vectors,
     for (size_t i = 0; i < first_place_in_ground_truth_feature_sets.size() - 1;
 	 ++i) {
 	gradient[i] -= first_place_in_ground_truth_feature_sets[i];
-
     }
 
     return gradient;
 }
 
 static void
-updateParameters(vector<double> & new_parameters, vector<double> & gradient,
-		 double & learning_rate) {
+update_parameters(vector<double>& new_parameters,
+		  const vector<double>& gradient,
+		  const double& learning_rate)
```

Comment:
Passing a primitive type by `const` reference is rarely sensible.  A `double` can fit in a CPU register on any modern CPU, so copying it should be cheap.  Similarly below.

(Passing it by a non-`const` reference can be useful if you want to return a value via a parameter, but we don't here).